src/stratigraphy/groundwater/utility.py

```diff
@@ -45,12 +45,15 @@ def extract_depth(text: str, max_depth: int) -> float | None:
     for pattern in depth_patterns:
         depth_match = regex.search(pattern, corrected_text)
         if depth_match:
-            depth = float(depth_match.group(1).replace(",", "."))
-            if depth > max_depth:
-                # If the extracted depth is greater than the max depth, set it to None and continue searching.
-                depth = None
-            else:
-                break
+            try:
+                depth = float(depth_match.group(1).replace(",", "."))
+                if depth > max_depth:
+                    # If the extracted depth is greater than the max depth, set it to None and continue searching.
+                    depth = None
+                else:
+                    break
+            except ValueError:
+                continue
```

Comment:
The code was essentially testing whether the year in tha date has two digits or four digits and then converted it into a date based format. We would run into the scenario of raising an exeception if the date has two digits to indicate the year instead of four. Since this implementation was not clear I changed it to something that is easier to read.

Reply:
@stijnvermeeren-swisstopo and @dcleres looked into it together, and we decided to roll back to the current version in the `main` branch and create a follow-up ticket (LGVISIUM-100) to address this issue.